src/curve_fit.jl

```diff
@@ -1,23 +1,29 @@
 immutable LsqFitResult{T,N}
     # simple type container for now, but can be expanded later
+    model::Function
     dof::Int
     param::Vector{T}
     resid::Vector{T}
+    rsquared::Float64
+    stderror::Float64
     jacobian::Matrix{T}
     converged::Bool
     wt::Array{T,N}
 end
 
 # provide a method for those who have their own Jacobian function
-function lmfit(f::Function, g::Function, p0, wt; kwargs...)
+function lmfit(model::Function, xpts::AbstractArray, ydata::AbstractArray, f::Function, g::Function, p0, wt; kwargs...)
     results = levenberg_marquardt(f, g, p0; kwargs...)
     p = minimizer(results)
     resid = f(p)
+    prediction = model(xpts, p)
+    rsquared = sum((prediction - mean(ydata)).^2) / sum((ydata - mean(ydata)).^2)
+    stderror = sqrt(sum((prediction - resid).^2) / length(prediction))
```

Comment:
I think this should be `stderror = sqrt(rsquared/dof)` instead, like [stdfit from gnuplot](http://gnuplot.sourceforge.net/docs_4.2/node86.html).